bpf/http_sock.h

```diff
@@ -65,15 +65,16 @@ static __always_inline bool is_http(char *p, u32 len, u8 *packet_type) {
         ((p[0] == 'P') && (p[1] == 'U') && (p[2] == 'T') && (p[3] == ' ') && (p[4] == '/')) ||                                                      // PUT
         ((p[0] == 'P') && (p[1] == 'A') && (p[2] == 'T') && (p[3] == 'C') && (p[4] == 'H') && (p[5] == ' ') && (p[6] == '/')) ||                    // PATCH
         ((p[0] == 'D') && (p[1] == 'E') && (p[2] == 'L') && (p[3] == 'E') && (p[4] == 'T') && (p[5] == 'E') && (p[6] == ' ') && (p[7] == '/')) ||   // DELETE
-        ((p[0] == 'H') && (p[1] == 'E') && (p[2] == 'A') && (p[3] == 'D') && (p[4] == ' ') && (p[5] == '/'))                                        // HEAD
+        ((p[0] == 'H') && (p[1] == 'E') && (p[2] == 'A') && (p[3] == 'D') && (p[4] == ' ') && (p[5] == '/')) ||                                     // HEAD
+        ((p[0] == 'O') && (p[1] == 'P') && (p[2] == 'T') && (p[3] == 'I') && (p[4] == 'O') && (p[5] == 'N') && (p[6] == 'S') && (p[7] == ' ') && (p[8] == '/'))   // OPTIONS
```

Comment:
I had to add options here, not for completeness but for an edge case scenario. Imagine we had a request that failed from a server without TCP close. We would technically have this data and we want to send it, perhaps 500, but previous to this PR we will never send a request unless we saw TCP close.

With this PR we detect that on a new request we have the same connection info from before, fully complete with response but we never sent it because the connection didn't close. The new request now pushes the old request and sets the new info. Without handling options we could perhaps mistakenly create fake OPTIONS event if we didn't close earlier connection, since Options wasn't recognized as a request until now.